.github/workflows/test.yml

```diff
@@ -0,0 +1,27 @@
+name: Run pytest on PR
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.11  # Choose your Python version
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt 
+        pip install -r requirements-dev.txt
```

Comment:
nit: Can separate steps for dep installation to improve clarity and simpler debugging.

```yaml
    - name: Install main dependencies
      run: |
        python -m pip install --upgrade pip
        pip install -r requirements.txt

    - name: Install development dependencies
      run: pip install -r requirements-dev.txt
```